visualization/include/visualization/xml_node.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+#include <utility>
+#include <string>
+
+namespace visualization::render {
+
+    class XMLNode {
+    public:
+        class Builder;
+
+        const std::vector<std::pair<std::string, std::string>> get_attributes() const;
+        const std::string get_tag() const;
+
+    private: 
+        XMLNode(const std::string tag, const std::vector<std::pair<std::string, std::string>> attributes);
+        std::string tag;
+        std::vector<std::pair<std::string, std::string>> attributes;
+    };
+
+    class XMLNode::Builder {
+    public: 
+        Builder& has_tag(const std::string name);
```

Comment:
`has_tag` sounds like a check, `with_tag` would be better in my opinion

Reply:
also what's a tag?

Reply:
After research, some sources name it "tag" (`<"tag">`) and other say "a tag has a name" (`"<name>"`). So, i prefer a tag because "name" is less clear.